From e452acabfcfa1159360ba72f907a12c651f4aeb6 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Wed, 28 Jul 2021 20:02:51 +0300
Subject: [PATCH] [SW-45031] gaudi2: avoid onix specific duplication

In this patch, Gaudi2 onix HL_TARGET specific
common code moved to common dir.

Change-Id: If4da6e0d47c39beb48b05b56d7e6db6d976935f5
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/arc/onix.c b/arch/gaudi2/arc/onix.c
index e08debd..4643ccb 100644
--- a/arch/gaudi2/arc/onix.c
+++ b/arch/gaudi2/arc/onix.c
@@ -5,40 +5,14 @@
  *
  */
 
-#include <stdbool.h>
 
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <cmd_module.h>
 #include <efuse.h>
-
-#include <psoc_global_conf_regs.h>
 
 #include "common.h"
 
-#define SYS_CLOCK_FREQUENCY	35000000ULL
-#define UART_CLOCK_FREQUENCY	SYS_CLOCK_FREQUENCY
-#define PLDM_UART_BAUDRATE	57600
 
 #define EFUSE_SYSCFG_ONIX_DEFAULT	0
-
-u32 hl_skip_pcie_init(void)
-{
-	return 1;
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_uart_init(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
-}
 
 u32 hl_spi_baudr_get(void)
 {
@@ -60,16 +34,4 @@
 	return ret;
 }
 
-enum cmd_check_res dev_specific_module_pci_check(void)
-{
-	/*
-	 * FLR and Hot Reset Registers are not part of PSOC and hence
-	 * writing to them results in ERROR that is resetting the
-	 * ONIX. To avoid it, simply return with NOT FOUND
-	 */
-	return NOT_FOUND;
-}
-
 void preboot_init_pcie_phy(void) {}
-
-void hl_dev_ts_init(void) {}
diff --git a/arch/gaudi2/arm/onix.c b/arch/gaudi2/arm/onix.c
index 4c28fb1..a0ef4b4 100644
--- a/arch/gaudi2/arm/onix.c
+++ b/arch/gaudi2/arm/onix.c
@@ -5,32 +5,10 @@
  *
  */
 
-#include <stdbool.h>
 
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <cmd_module.h>
-#include <psoc_global_conf_regs.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
-
-u32 hl_skip_pcie_init(void)
-{
-	return 1;
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_uart_init(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(SYS_CLOCK_FREQUENCY / (HL_UART_BAUDRATE / 2));
-}
 
 u32 hl_spi_baudr_get(void)
 {
@@ -42,16 +20,4 @@
 	return (factor + 1) & 0xFFFE;
 }
 
-enum cmd_check_res dev_specific_module_pci_check(void)
-{
-	/*
-	 * FLR and Hot Reset Registers are not part of PSOC and hence
-	 * writing to them results in ERROR that is resetting the
-	 * ONIX. To avoid it, simply return with NOT FOUND
-	 */
-	return NOT_FOUND;
-}
-
 void preboot_sync_btl(void) {}
-
-void hl_dev_ts_init(void) {}
diff --git a/arch/gaudi2/common/onix.c b/arch/gaudi2/common/onix.c
index e76553e..2e0cdf2 100644
--- a/arch/gaudi2/common/onix.c
+++ b/arch/gaudi2/common/onix.c
@@ -5,4 +5,40 @@
  *
  */
 
-/* TODO: Gaudi2 onix(HL_TARGET) specific common functions to be moved here */
+#include <preboot.h>
+#include <psoc_global_conf_regs.h>
+#include <cmd_module.h>
+#include <uart.h>
+
+
+#define SYS_CLOCK_FREQUENCY	35000000ULL
+
+
+u32 hl_skip_pcie_init(void)
+{
+	return 1;
+}
+
+u64 hl_pll_timestamp_freq_get(void)
+{
+	return SYS_CLOCK_FREQUENCY;
+}
+
+void hl_uart_init(void)
+{
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
+	preboot_uart_init(SYS_CLOCK_FREQUENCY / (HL_UART_BAUDRATE / 2));
+}
+
+enum cmd_check_res dev_specific_module_pci_check(void)
+{
+	/*
+	 * FLR and Hot Reset Registers are not part of PSOC and hence
+	 * writing to them results in ERROR that is resetting the
+	 * ONIX. To avoid it, simply return with NOT FOUND
+	 */
+	return NOT_FOUND;
+}
+
+void hl_dev_ts_init(void) {}
